src/autolabel/dataset/dataset.py

```diff
@@ -0,0 +1,263 @@
+from typing import Dict, List, Union
+import pandas as pd
+from autolabel.configs import AutolabelConfig
+from autolabel.dataset.validation import TaskDataValidation
+from autolabel.schema import MetricResult, LLMAnnotation
+from tabulate import tabulate
+import logging
+from autolabel.utils import print_table
+from rich.console import Console
+import json
+from autolabel.tasks import TaskFactory
+
+logger = logging.getLogger(__name__)
+
+METRIC_TABLE_STYLE = "cyan bold"
+
+
+class AutolabelDataset:
+    """Data Attributes"""
+
+    inputs: List[Dict]
+    df: pd.DataFrame
+    gt_labels: List
+    config: AutolabelConfig
+
+    class Config:
+        arbitrary_types_allowed = True
+
+    def __init__(
+        self,
+        dataset: Union[pd.DataFrame, str],
+        config: AutolabelConfig,
+        max_items: int = None,
+        start_index: int = 0,
```

Comment:
i think these should still belong to run/plan/transform methods ? 

feels more like a runtime parameter rather than something needed to create the dataset itself - 

in practice, moving this to the dataset would mean the user needs to create a new dataset if they want to run labeling using the same config/base csv but max_items = 100, vs max_items = 1000